components/brave_wallet/browser/keyring_controller.cc

```diff
@@ -635,6 +655,144 @@ void KeyringController::GetPrivateKeyForDefaultKeyringAccount(
   std::move(callback).Run(!private_key.empty(), private_key);
 }
 
+bool KeyringController::IsFilecoinAccount(const std::string& account) const {
+  for (const auto& filecoin_account_info :
+       GetImportedAccountsForKeyring(prefs_, kFilecoinKeyringId)) {
+    mojom::AccountInfoPtr account_info = mojom::AccountInfo::New();
+    if (account == filecoin_account_info.account_address) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void KeyringController::ImportFilecoinSECP256K1Account(
+    const std::string& account_name,
+    const std::string& private_key_hex,
+    const std::string& network,
+    ImportFilecoinSECP256K1AccountCallback callback) {
+  if (!IsFilecoinEnabled()) {
+    VLOG(1) << "Filecoin feature is not enabled";
+    return;
+  }
+  if (account_name.empty() || private_key_hex.empty() || !encryptor_) {
+    std::move(callback).Run(false, "");
+    return;
+  }
+  std::vector<uint8_t> private_key;
+  if (!base::HexStringToBytes(private_key_hex, &private_key)) {
+    LOG(ERROR) << private_key_hex << " result size: " << private_key.size();
```

Comment:
Is it safe to log the private key here?

Reply:
don't log the private key, there's a chance a user might submit a log one day and not realize it.

Reply:
done